In [1]:
import pandas as pd
import numpy as np

In [2]:
allergy = pd.read_excel('C:/Users/kesle/OneDrive/Desktop/food-allergy-analysis-Zenodo-final.xlsx')

In [3]:
allergy.head()

SUBJECTID  BIRTHYEAR GENDERFACTOR    RACEFACTOR    ETHNICITYFACTOR  \
0          1       2006  S1 - Female    R1 - Black  E0 - Non-Hispanic   
1          2       1994  S1 - Female    R0 - White  E0 - Non-Hispanic   
2          3       2006    S0 - Male    R0 - White      E1 - Hispanic   
3          4       2004    S0 - Male  R4 - Unknown      E1 - Hispanic   
4          5       2006  S1 - Female    R1 - Black  E0 - Non-Hispanic   

   AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  SOYALGSTART  \
0       0.093087                NaN           NaN           NaN          NaN   
1      12.232717                NaN           NaN           NaN          NaN   
2       0.010951                NaN           NaN      1.002053          NaN   
3       2.398357                NaN           NaN           NaN          NaN   
4       0.013689                NaN           NaN           NaN          NaN   

   EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  SESAMEALGSTART  \
0          NaN            NaN             NaN             NaN   
1          NaN            NaN             NaN             NaN   
2          NaN            NaN             NaN             NaN   
3          NaN            NaN             NaN             NaN   
4          NaN            NaN             NaN             NaN   

   ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  FIRSTASTHMARX  \
0              NaN                    NaN          NaN            NaN   
1              NaN                    NaN          NaN      12.262834   
2         4.884326               3.917864     5.127995       1.404517   
3              NaN                    NaN          NaN            NaN   
4              NaN                    NaN          NaN            NaN   

   NUMASTHMARX  
0          NaN  
1          2.0  
2          4.0  
3          NaN  
4          NaN

#Viewing RACEFACTOR value counts

In [4]:
allergy['RACEFACTOR'].value_counts()

R0 - White                        183308
R1 - Black                         97795
R4 - Unknown                       40940
R2 - Asian or Pacific Islander      9152
R3 - Other                          2005
Name: RACEFACTOR, dtype: int64

#Will use 0 for White, 1 for Black, 2 for Asian or Pacific Islander, 3 for Other, 4 for Unknown.

In [5]:
def RACEFACTOR_recode (series):
    if series == "R0 - White":
        return 0
    if series == "R1 - Black":
        return 1
    if series == "R2 - Asian or Pacific Islander":
        return 2
    if series == "R3 - Other":
        return 3
    if series == "R4 - Unknown":
        return 4

In [6]:
allergy['RACEFACTOR']=allergy['RACEFACTOR'].apply(RACEFACTOR_recode)

In [7]:
allergy.head()

SUBJECTID  BIRTHYEAR GENDERFACTOR  RACEFACTOR    ETHNICITYFACTOR  \
0          1       2006  S1 - Female           1  E0 - Non-Hispanic   
1          2       1994  S1 - Female           0  E0 - Non-Hispanic   
2          3       2006    S0 - Male           0      E1 - Hispanic   
3          4       2004    S0 - Male           4      E1 - Hispanic   
4          5       2006  S1 - Female           1  E0 - Non-Hispanic   

   AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  SOYALGSTART  \
0       0.093087                NaN           NaN           NaN          NaN   
1      12.232717                NaN           NaN           NaN          NaN   
2       0.010951                NaN           NaN      1.002053          NaN   
3       2.398357                NaN           NaN           NaN          NaN   
4       0.013689                NaN           NaN           NaN          NaN   

   EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  SESAMEALGSTART  \
0          NaN            NaN             NaN             NaN   
1          NaN            NaN             NaN             NaN   
2          NaN            NaN             NaN             NaN   
3          NaN            NaN             NaN             NaN   
4          NaN            NaN             NaN             NaN   

   ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  FIRSTASTHMARX  \
0              NaN                    NaN          NaN            NaN   
1              NaN                    NaN          NaN      12.262834   
2         4.884326               3.917864     5.127995       1.404517   
3              NaN                    NaN          NaN            NaN   
4              NaN                    NaN          NaN            NaN   

   NUMASTHMARX  
0          NaN  
1          2.0  
2          4.0  
3          NaN  
4          NaN

#Success. Now onto GENDERFACTOR

In [8]:
allergy['GENDERFACTOR'].value_counts()

S0 - Male      169032
S1 - Female    164168
Name: GENDERFACTOR, dtype: int64

#Will use 0 for Male, 1 for Female.

In [9]:
def GENDERFACTOR_recode (series):
    if series == "S0 - Male":
        return 0
    if series == "S1 - Female":
        return 1

In [10]:
allergy['GENDERFACTOR']=allergy['GENDERFACTOR'].apply(GENDERFACTOR_recode)

In [11]:
allergy.head()

SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR    ETHNICITYFACTOR  \
0          1       2006             1           1  E0 - Non-Hispanic   
1          2       1994             1           0  E0 - Non-Hispanic   
2          3       2006             0           0      E1 - Hispanic   
3          4       2004             0           4      E1 - Hispanic   
4          5       2006             1           1  E0 - Non-Hispanic   

   AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  SOYALGSTART  \
0       0.093087                NaN           NaN           NaN          NaN   
1      12.232717                NaN           NaN           NaN          NaN   
2       0.010951                NaN           NaN      1.002053          NaN   
3       2.398357                NaN           NaN           NaN          NaN   
4       0.013689                NaN           NaN           NaN          NaN   

   EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  SESAMEALGSTART  \
0          NaN            NaN             NaN             NaN   
1          NaN            NaN             NaN             NaN   
2          NaN            NaN             NaN             NaN   
3          NaN            NaN             NaN             NaN   
4          NaN            NaN             NaN             NaN   

   ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  FIRSTASTHMARX  \
0              NaN                    NaN          NaN            NaN   
1              NaN                    NaN          NaN      12.262834   
2         4.884326               3.917864     5.127995       1.404517   
3              NaN                    NaN          NaN            NaN   
4              NaN                    NaN          NaN            NaN   

   NUMASTHMARX  
0          NaN  
1          2.0  
2          4.0  
3          NaN  
4          NaN

#Done. Now onto ETHNICITYFACTOR.

In [12]:
allergy['ETHNICITYFACTOR'].value_counts()

E0 - Non-Hispanic    317868
E1 - Hispanic         15332
Name: ETHNICITYFACTOR, dtype: int64

#Will use 0 for Non-Hispanic, 1 for Hispanic.

In [13]:
def ETHNICITYFACTOR_recode (series):
    if series == "E0 - Non-Hispanic":
        return 0
    if series == "E1 - Hispanic":
        return 1

In [14]:
allergy['ETHNICITYFACTOR']=allergy['ETHNICITYFACTOR'].apply(ETHNICITYFACTOR_recode)

In [15]:
allergy.head()

SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  ETHNICITYFACTOR  \
0          1       2006             1           1                0   
1          2       1994             1           0                0   
2          3       2006             0           0                1   
3          4       2004             0           4                1   
4          5       2006             1           1                0   

   AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  SOYALGSTART  \
0       0.093087                NaN           NaN           NaN          NaN   
1      12.232717                NaN           NaN           NaN          NaN   
2       0.010951                NaN           NaN      1.002053          NaN   
3       2.398357                NaN           NaN           NaN          NaN   
4       0.013689                NaN           NaN           NaN          NaN   

   EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  SESAMEALGSTART  \
0          NaN            NaN             NaN             NaN   
1          NaN            NaN             NaN             NaN   
2          NaN            NaN             NaN             NaN   
3          NaN            NaN             NaN             NaN   
4          NaN            NaN             NaN             NaN   

   ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  FIRSTASTHMARX  \
0              NaN                    NaN          NaN            NaN   
1              NaN                    NaN          NaN      12.262834   
2         4.884326               3.917864     5.127995       1.404517   
3              NaN                    NaN          NaN            NaN   
4              NaN                    NaN          NaN            NaN   

   NUMASTHMARX  
0          NaN  
1          2.0  
2          4.0  
3          NaN  
4          NaN

#Now to drop NA variables

In [16]:
allergy.fillna(0)

SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  ETHNICITYFACTOR  \
0               1       2006             1           1                0   
1               2       1994             1           0                0   
2               3       2006             0           0                1   
3               4       2004             0           4                1   
4               5       2006             1           1                0   
...           ...        ...           ...         ...              ...   
333195     333196       2006             0           0                0   
333196     333197       2006             1           1                0   
333197     333198       2006             0           0                0   
333198     333199       2006             0           3                0   
333199     333200       2006             1           0                0   

        AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  \
0            0.093087                0.0           0.0      0.000000   
1           12.232717                0.0           0.0      0.000000   
2            0.010951                0.0           0.0      1.002053   
3            2.398357                0.0           0.0      0.000000   
4            0.013689                0.0           0.0      0.000000   
...               ...                ...           ...           ...   
333195       0.736482                0.0           0.0      0.000000   
333196       0.019165                0.0           0.0      0.000000   
333197       0.443532                0.0           0.0      0.000000   
333198       0.013689                0.0           0.0      0.000000   
333199       0.019165                0.0           0.0      0.000000   

        SOYALGSTART  EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  \
0               0.0          0.0            0.0             0.0   
1               0.0          0.0            0.0             0.0   
2               0.0          0.0            0.0             0.0   
3               0.0          0.0            0.0             0.0   
4               0.0          0.0            0.0             0.0   
...             ...          ...            ...             ...   
333195          0.0          0.0            0.0             0.0   
333196          0.0          0.0            0.0             0.0   
333197          0.0          0.0            0.0             0.0   
333198          0.0          0.0            0.0             0.0   
333199          0.0          0.0            0.0             0.0   

        SESAMEALGSTART  ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  \
0                  0.0         0.000000               0.000000     0.000000   
1                  0.0         0.000000               0.000000     0.000000   
2                  0.0         4.884326               3.917864     5.127995   
3                  0.0         0.000000               0.000000     0.000000   
4                  0.0         0.000000               0.000000     0.000000   
...                ...              ...                    ...          ...   
333195             0.0         0.736482               0.000000     5.426420   
333196             0.0         0.000000               0.000000     0.000000   
333197             0.0         0.000000               0.000000     2.321697   
333198             0.0         0.000000               0.000000     0.000000   
333199             0.0         0.347707               0.000000     0.000000   

        FIRSTASTHMARX  NUMASTHMARX  
0            0.000000          0.0  
1           12.262834          2.0  
2            1.404517          4.0  
3            0.000000          0.0  
4            0.000000          0.0  
...               ...          ...  
333195       0.000000          0.0  
333196       0.000000          0.0  
333197       2.321697          4.0  
333198       0.000000          0.0  
333199       0.000000          0.0  

[333200 rows x 19 columns]

#recode age

In [17]:
def Age (series):
    if series <= 2:
        return  "2"
    if series <= 4:
        return "4"
    if series <= 6:
        return "6"
    if series <= 10:
        return "10"
    else:
        return "11 - 17"
    
allergy["StartAge"] = allergy['AGESTARTYEARS'].apply(Age)   

In [18]:
allergy["StartAge"].value_counts()

2          170240
10          49896
11 - 17     49809
4           33216
6           30039
Name: StartAge, dtype: int64

#We are going to be focusing on age 2 and under for our analysis. Deleting any values that are not on that range.

In [19]:
allergy = allergy[allergy['StartAge'].isin(['2'])]

In [20]:
allergy.head()

SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  ETHNICITYFACTOR  \
0          1       2006             1           1                0   
2          3       2006             0           0                1   
4          5       2006             1           1                0   
5          6       2006             0           0                0   
6          7       2006             1           0                0   

   AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  SOYALGSTART  \
0       0.093087                NaN           NaN           NaN          NaN   
2       0.010951                NaN           NaN      1.002053          NaN   
4       0.013689                NaN           NaN           NaN          NaN   
5       0.087611                NaN           NaN           NaN          NaN   
6       0.010951                NaN           NaN           NaN          NaN   

   EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  SESAMEALGSTART  \
0          NaN            NaN             NaN             NaN   
2          NaN            NaN             NaN             NaN   
4          NaN            NaN             NaN             NaN   
5          NaN            NaN             NaN             NaN   
6          NaN            NaN             NaN             NaN   

   ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  FIRSTASTHMARX  \
0              NaN                    NaN          NaN            NaN   
2         4.884326               3.917864     5.127995       1.404517   
4              NaN                    NaN          NaN            NaN   
5              NaN                    NaN          NaN            NaN   
6              NaN                    NaN          NaN            NaN   

   NUMASTHMARX StartAge  
0          NaN        2  
2          4.0        2  
4          NaN        2  
5          NaN        2  
6          NaN        2

In [21]:
print(allergy)

        SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  ETHNICITYFACTOR  \
0               1       2006             1           1                0   
2               3       2006             0           0                1   
4               5       2006             1           1                0   
5               6       2006             0           0                0   
6               7       2006             1           0                0   
...           ...        ...           ...         ...              ...   
333195     333196       2006             0           0                0   
333196     333197       2006             1           1                0   
333197     333198       2006             0           0                0   
333198     333199       2006             0           3                0   
333199     333200       2006             1           0                0   

        AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  \
0            0.093087      

In [22]:
allergy.fillna(0)

SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  ETHNICITYFACTOR  \
0               1       2006             1           1                0   
2               3       2006             0           0                1   
4               5       2006             1           1                0   
5               6       2006             0           0                0   
6               7       2006             1           0                0   
...           ...        ...           ...         ...              ...   
333195     333196       2006             0           0                0   
333196     333197       2006             1           1                0   
333197     333198       2006             0           0                0   
333198     333199       2006             0           3                0   
333199     333200       2006             1           0                0   

        AGESTARTYEARS  SHELLFISHALGSTART  FISHALGSTART  MILKALGSTART  \
0            0.093087                0.0           0.0      0.000000   
2            0.010951                0.0           0.0      1.002053   
4            0.013689                0.0           0.0      0.000000   
5            0.087611                0.0           0.0      0.000000   
6            0.010951                0.0           0.0      0.000000   
...               ...                ...           ...           ...   
333195       0.736482                0.0           0.0      0.000000   
333196       0.019165                0.0           0.0      0.000000   
333197       0.443532                0.0           0.0      0.000000   
333198       0.013689                0.0           0.0      0.000000   
333199       0.019165                0.0           0.0      0.000000   

        SOYALGSTART  EGGALGSTART  WHEATALGSTART  PEANUTALGSTART  \
0               0.0          0.0            0.0             0.0   
2               0.0          0.0            0.0             0.0   
4               0.0          0.0            0.0             0.0   
5               0.0          0.0            0.0             0.0   
6               0.0          0.0            0.0             0.0   
...             ...          ...            ...             ...   
333195          0.0          0.0            0.0             0.0   
333196          0.0          0.0            0.0             0.0   
333197          0.0          0.0            0.0             0.0   
333198          0.0          0.0            0.0             0.0   
333199          0.0          0.0            0.0             0.0   

        SESAMEALGSTART  ATOPICDERMSTART  ALLERGICRHINITISSTART  ASTHMASTART  \
0                  0.0         0.000000               0.000000     0.000000   
2                  0.0         4.884326               3.917864     5.127995   
4                  0.0         0.000000               0.000000     0.000000   
5                  0.0         0.000000               0.000000     0.000000   
6                  0.0         0.000000               0.000000     0.000000   
...                ...              ...                    ...          ...   
333195             0.0         0.736482               0.000000     5.426420   
333196             0.0         0.000000               0.000000     0.000000   
333197             0.0         0.000000               0.000000     2.321697   
333198             0.0         0.000000               0.000000     0.000000   
333199             0.0         0.347707               0.000000     0.000000   

        FIRSTASTHMARX  NUMASTHMARX StartAge  
0            0.000000          0.0        2  
2            1.404517          4.0        2  
4            0.000000          0.0        2  
5            0.000000          0.0        2  
6            0.000000          0.0        2  
...               ...          ...      ...  
333195       0.000000          0.0        2  
333196       0.000000          0.0        2  
333197       2.321697          4.0        2  
333198       0.000000       

#Rerun the correlation matrix with our selected data.

In [23]:
allergy.corr(method='pearson')

C:\Users\kesle\AppData\Local\Temp\ipykernel_17256\180886369.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  allergy.corr(method='pearson')


SUBJECTID  BIRTHYEAR  GENDERFACTOR  RACEFACTOR  \
SUBJECTID               1.000000  -0.489062      0.001230   -0.080132   
BIRTHYEAR              -0.489062   1.000000     -0.001011    0.114520   
GENDERFACTOR            0.001230  -0.001011      1.000000    0.015679   
RACEFACTOR             -0.080132   0.114520      0.015679    1.000000   
ETHNICITYFACTOR        -0.034193   0.059561      0.002151    0.254160   
AGESTARTYEARS           0.020294  -0.396665     -0.006039   -0.003657   
SHELLFISHALGSTART       0.239545  -0.664759      0.015438   -0.109318   
FISHALGSTART            0.244603  -0.544438      0.024322   -0.129044   
MILKALGSTART            0.169476  -0.492001     -0.025640   -0.046069   
SOYALGSTART             0.207956  -0.575459     -0.053627   -0.030910   
EGGALGSTART             0.127065  -0.380680      0.006849   -0.018100   
WHEATALGSTART           0.177158  -0.565883      0.020487   -0.071897   
PEANUTALGSTART          0.178380  -0.535932      0.028729   -0.008546   
SESAMEALGSTART          0.280119  -0.731301     -0.028592   -0.123608   
ATOPICDERMSTART         0.135692  -0.441166      0.052902   -0.068849   
ALLERGICRHINITISSTART   0.127048  -0.524965      0.052116   -0.021974   
ASTHMASTART             0.106450  -0.404068      0.065272   -0.048653   
FIRSTASTHMARX           0.163686  -0.554295      0.049957   -0.084978   
NUMASTHMARX             0.076148  -0.174777     -0.044633    0.013170   

                       ETHNICITYFACTOR  AGESTARTYEARS  SHELLFISHALGSTART  \
SUBJECTID                    -0.034193       0.020294           0.239545   
BIRTHYEAR                     0.059561      -0.396665          -0.664759   
GENDERFACTOR                  0.002151      -0.006039           0.015438   
RACEFACTOR                    0.254160      -0.003657          -0.109318   
ETHNICITYFACTOR               1.000000      -0.003814          -0.040192   
AGESTARTYEARS                -0.003814       1.000000           0.274471   
SHELLFISHALGSTART            -0.040192       0.274471           1.000000   
FISHALGSTART                 -0.031569       0.243136           0.663456   
MILKALGSTART                 -0.027055       0.347761           0.523699   
SOYALGSTART                  -0.003021       0.346532           0.532329   
EGGALGSTART                   0.000931       0.234535           0.544851   
WHEATALGSTART                 0.047961       0.342067           0.586116   
PEANUTALGSTART                0.016429       0.273706           0.559735   
SESAMEALGSTART               -0.005276       0.317998           0.647249   
ATOPICDERMSTART               0.000764       0.334368           0.383358   
ALLERGICRHINITISSTART        -0.004864       0.269738           0.447553   
ASTHMASTART                  -0.019087       0.260462           0.311876   
FIRSTASTHMARX                -0.038352       0.323648           0.400899   
NUMASTHMARX                   0.001569       0.073576           0.110805   

                       FISHALGSTART  MILKALGSTART  SOYALGSTART  EGGALGSTART  \
SUBJECTID                  0.244603      0.169476     0.207956     0.127065   
BIRTHYEAR                 -0.544438     -0.492001    -0.575459    -0.380680   
GENDERFACTOR               0.024322     -0.025640    -0.053627     0.006849   
RACEFACTOR                -0.129044     -0.046069    -0.030910    -0.018100   
ETHNICITYFACTOR           -0.031569     -0.027055    -0.003021     0.000931   
AGESTARTYEARS              0.243136      0.347761     0.346532     0.234535   
SHELLFISHALGSTART          0.663456      0.523699     0.532329     0.544851   
FISHALGSTART               1.000000      0.505942     0.463320     0.513258   
MILKALGSTART               0.505942      1.000000     0.845943     0.340947   
SOYALGSTART                0.463320      0.845943     1.000000     0.732030   
EGGALGSTART                0.513258      0.340947     0.732030     1.000000   
WHEATALGSTART              0.409735      0.857956     0.923160     0.790604   
PEANUTALGSTART

#Looks good! Now to get the means for the columns.

In [24]:
allergy.AGESTARTYEARS.describe()

count    170240.000000
mean          0.261607
std           0.483896
min          -4.312115
25%           0.013689
50%           0.021903
75%           0.180698
max           1.998631
Name: AGESTARTYEARS, dtype: float64

#Mean age for diagnosis is around 2 months, with a standard deviation of 5 months(rounding up.)

In [25]:
allergy.SHELLFISHALGSTART.describe()

count    2338.000000
mean        4.524523
std         2.927695
min         0.093087
25%         2.094456
50%         3.787817
75%         6.416838
max        13.930185
Name: SHELLFISHALGSTART, dtype: float64

#Mean for shellfish allergy diagnosis is around 4.5 with standard deviation of 3(rounding up)

In [26]:
allergy.FISHALGSTART.describe()

count    1000.000000
mean        3.420734
std         2.500149
min         0.076660
25%         1.532512
50%         2.569473
75%         4.619439
max        13.982204
Name: FISHALGSTART, dtype: float64

#Mean for fish allergy diagnosis is 3 years. Standard deviation of 2.5.

In [27]:
allergy.MILKALGSTART.describe()

count    5647.000000
mean        1.700579
std         2.053006
min       -60.112252
25%         0.555784
50%         1.070500
75%         2.012320
max        13.127995
Name: MILKALGSTART, dtype: float64

#Mean of milk allergy diagnosis is around 1.8 years. Standard deviation is just over 2. This is an interesting point. Most children under 1 year old will drink breast milk or formula. Interesting to consider that most milk allergies are going to be diagnosed during the second year of the child's life, as around one year of age, children switch to drinking whole milk instead of breast milk or formula.

In [28]:
allergy.SOYALGSTART.describe()

count    1829.000000
mean        2.086602
std         2.261067
min         0.027379
25%         0.635181
50%         1.204654
75%         2.609172
max        13.095140
Name: SOYALGSTART, dtype: float64

#Mean of soy allergy diagnosis is around 2 years old. Standard deviation is around 2, as well.

In [29]:
allergy.EGGALGSTART.describe()

count    4719.000000
mean        1.995404
std         2.249677
min       -98.866530
25%         1.002053
50%         1.344285
75%         2.449008
max        13.218344
Name: EGGALGSTART, dtype: float64

#Mean of egg allergy diagnosis is around 2(rounding up). Standard deviation of around 2, as well.

In [30]:
allergy.WHEATALGSTART.describe()

count    796.000000
mean       2.444340
std        2.434673
min        0.156057
25%        0.864476
50%        1.430527
75%        3.088980
max       14.138261
Name: WHEATALGSTART, dtype: float64

#Mean of wheat allergy diagnosis is around 2. standard deviation is approx. same.

In [31]:
allergy.PEANUTALGSTART.describe()

count    5576.000000
mean        2.783521
std         2.141779
min         0.021903
25%         1.264887
50%         2.021903
75%         3.650240
max        13.875428
Name: PEANUTALGSTART, dtype: float64

#Mean of peanut allergy diagnosis is after 2.5. standard deviation is lower, at 2.

In [32]:
allergy.SESAMEALGSTART.describe()

count    524.000000
mean       4.435120
std        2.827863
min        0.487337
25%        2.063655
50%        3.845311
75%        6.134839
max       12.906229
Name: SESAMEALGSTART, dtype: float64

#Interesting. Mean of sesame allergy diagnosis is after the age of 4. standard deviation is closer to 2.5.

In [33]:
allergy.ATOPICDERMSTART.describe()

count    34573.000000
mean         1.970353
std          2.457132
min       -103.961670
25%          0.435318
50%          1.133470
75%          2.666667
max         13.894593
Name: ATOPICDERMSTART, dtype: float64

#Mean of eczema diagnosis is close to 2, if rounding up. Standard deviation is around 2.5(rounding up)

In [34]:
allergy.ALLERGICRHINITISSTART.describe()

count    24610.000000
mean         4.296696
std          2.464025
min         -2.642026
25%          2.368241
50%          3.882272
75%          5.752225
max         14.026010
Name: ALLERGICRHINITISSTART, dtype: float64

#Mean of Rhinitis diagnosis is after children turn 4. Standard deviation of 2.5(rounding up)

In [35]:
allergy.ASTHMASTART.describe()

count    32615.000000
mean         3.075938
std          2.242110
min          1.002053
25%          1.319644
50%          2.360027
75%          4.113621
max         14.294319
Name: ASTHMASTART, dtype: float64

#Mean of asthma diagnosis is around 3. standard deviation of 2.

#Illness diagnosis means seem pretty interesting. They all appear to happen after the mean age of 2. Curious to see what allergies correlate with those diagnosis.

In [36]:
allergy.FIRSTASTHMARX.describe()

count    64697.000000
mean         3.174012
std          2.285018
min          1.002053
25%          1.437372
50%          2.335387
75%          4.240931
max         14.294319
Name: FIRSTASTHMARX, dtype: float64

#First asthma attack mean is around 3 years old. Standard deviation of 2.

In [37]:
allergy.NUMASTHMARX.describe()

count    64697.000000
mean         5.328377
std          6.504503
min          1.000000
25%          1.000000
50%          3.000000
75%          7.000000
max        109.000000
Name: NUMASTHMARX, dtype: float64

#Mean of asthma attacks per child is around 5. Standard deviation of 6.5.